YourFirstPR.md

```diff
@@ -0,0 +1,77 @@
+
+## Checking out the fastlane repo
+
+- Click the “fork” button in the upper right corner of the [main fastlane repo][fastlane]
+- Clone your fork:
+  - `git clone <YOUR_GITHUB_USER> git@github.com:<YOUR_GITHUB_USER>/fastlane.git`
+- Install dependencies:
+  - Run `bundle install` in the project root
+  - You also might need to run `bundle install` in each of the tool's subdirectories, e.g. `cd gym && bundle install`
+- Create a new branch to work on:
+  - `git checkout -b <YOUR_BRANCH_NAME>`
+  - A good name for a branch describes the thing you’ll be working on, e.g. `docs-fixes`, `fix-deliver-upload`, `gym-build-android-app`, etc.
+- That’s it! Now you’re ready to work on fastlane 
+
+## Testing your local changes
+
+Make sure to run the automated tests using `bundle exec` to ensure you’re running the correct version of `rspec` and `rubocop`
+
+First, navigate into the sub-directory of the tool you want to test.
+
+To run the automated unit tests
+
+```
+bundle exec rspec
+```
+
+To verify and auto-fix the code style
+
+```
+bundle exec rubocop -a
+```
+
+After introducing some changes to the _fastlane_ source code, you probably want to test the changes for your application. The easiest way to do so it use [bundler](https://bundler.io/).
+
+Copy the Gemfile [.assets/Gemfile](.assets/Gemfile) from your local fastlane clone and drop it into your project's root folder.
+
+Make sure to change the `local_fastlane_path` variable to point to your fastlane clone, e.g. `~/fastlane`, then you can run
+```
+bundle update
+```
+in your project’s root directory. After doing so, you can verify you’re using the local version by running
+
+```
+bundle show fastlane
+```
+
+which should print out the path to your local development environment.
+
+From now on, every time you introduce a change to your local _fastlane_ code base, you can immediately test it by running `bundle exec fastlane …`
+
+## Submitting the PR
+
+When the coding is done and you’re finished testing your changes, you are ready to submit the PR to the [fastlane main repo][fastlane]. Everything you need to know about submitting the PR itself is inside our [Pull Request Template][pr template]. Some best practices are:
+
+- Create a separate PR for each tool that you’ve worked on
+- Use a descriptive title 
+- Link the issues that are related to your PR in the body
+
+## After the review
+
+Once a core member has reviewed your PR, you might need to make changes before it gets merged. To make it easier on us, please make sure to avoid using `git commit --amend` and force pushes so that only the changes you did after the review will be visible on GitHub. Feel free to use whatever commit messages you like, as we will squash them anyway. When you are done addressing your review, also add a small comment like “Feedback addressed @<your_reviewer>”. 
```

Comment:
Suggestion:

"To make it easier on us... visible on GitHub." ->

"To make it easier on us, please make sure to avoid using `git commit --amend` or force pushes to make corrections. By avoiding rewriting the commit history, you will allow each round of edits to become its own visible commit. This helps the people who need to review your code easily understand exactly what has changed since the last time they looked."